Tests/CacheAdvanceTests/CacheAdvanceTests.swift

```diff
@@ -0,0 +1,391 @@
+        let message = "This is a test"
+        let cache = try CacheAdvance<String>(
+            file: testFileLocation,
+            maximumBytes: try requiredByteCount(for: [message], cacheWillRoll: false) - 1,
+            shouldOverwriteOldMessages: false)
+
+        XCTAssertThrowsError(try cache.append(message: message)) {
+            XCTAssertEqual($0 as? CacheAdvanceWriteError, CacheAdvanceWriteError.messageDataTooLarge)
+        }
+
+        let messages = try cache.messages()
+        XCTAssertEqual(messages, [])
+    }
+
+    func test_append_singleMessageThrowsIfDoesNotFitAndCacheRolls() throws {
+        let message = "This is a test"
+        let cache = try CacheAdvance<String>(
+            file: testFileLocation,
+            maximumBytes: try requiredByteCount(for: [message], cacheWillRoll: true) - 1,
+            shouldOverwriteOldMessages: true)
+
+        XCTAssertThrowsError(try cache.append(message: message)) {
+            XCTAssertEqual($0 as? CacheAdvanceWriteError, CacheAdvanceWriteError.messageDataTooLarge)
+        }
+
+        let messages = try cache.messages()
+        XCTAssertEqual(messages, [])
+    }
+
+    func test_append_multipleMessagesCanBeRetrieved() throws {
+        let cache = try CacheAdvance<String>(
+            file: testFileLocation,
+            maximumBytes: try requiredByteCount(for: lorumIpsumMessages, cacheWillRoll: false),
+            shouldOverwriteOldMessages: false)
+        for message in lorumIpsumMessages {
+            try cache.append(message: message)
+        }
+
+        let messages = try cache.messages()
+        XCTAssertEqual(messages, lorumIpsumMessages)
+    }
+
+    func test_append_dropsLastMessageIfCacheDoesNotRollAndLastMessageDoesNotFit() throws {
+        let cache = try CacheAdvance<String>(
+            file: testFileLocation,
+            maximumBytes: try requiredByteCount(for: lorumIpsumMessages, cacheWillRoll: false),
+            shouldOverwriteOldMessages: false)
+        for message in lorumIpsumMessages {
+            try cache.append(message: message)
+        }
+
+        XCTAssertThrowsError(try cache.append(message: "This message won't fit")) {
+            XCTAssertEqual($0 as? CacheAdvanceWriteError, CacheAdvanceWriteError.messageDataTooLarge)
+        }
+
+        let messages = try cache.messages()
+        XCTAssertEqual(messages, lorumIpsumMessages)
+    }
+
+    func test_append_dropsFirstMessageIfCacheRollsAndLastMessageDoesNotFitAndIsShorterThanFirstMessage() throws {
+        let cache = try CacheAdvance<String>(
+            file: testFileLocation,
+            maximumBytes: try requiredByteCount(for: lorumIpsumMessages, cacheWillRoll: true),
+            shouldOverwriteOldMessages: true)
+        for message in lorumIpsumMessages {
+            try cache.append(message: message)
+        }
+
+        // Append a message that is shorter than the first message in lorumIpsumMessages.
```

Comment:
Can we turn this comment into an `XCTAssert` or some sort?

Reply:
I think the name of the method should help enough. There’s only one assertion per test (with the exception of the methods expecting errors to be thrown), so I’m not sure how much an assertion comment helps.

Commenting explicitly what this line is doing I think does help with the long-term maintenance of this test.